src/test/java/com/bobocode/petros/scaner/AnnotationDependencyClassScannerTest.java

```diff
@@ -1,21 +1,98 @@
 package com.bobocode.petros.scaner;
 
+import com.bobocode.petros.container.DependencyDefinition;
+import com.bobocode.petros.exception.DefaultConstructorNotFoundException;
+import com.bobocode.petros.exception.MultipleInjectConstructorsException;
+import com.bobocode.petros.exception.NoSuchPackageFoundException;
+import com.bobocode.petros.testpackage.scan.dependency.SuccessfullyBuildDefinitionDependencyCase;
+import lombok.SneakyThrows;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
+import java.util.function.Predicate;
+
 class AnnotationDependencyClassScannerTest {
-    private static final String SOME_PACKAGE = "some.package";
+
+    private static final String TESTING_TEST_CLASSES_PATH = "target/test-classes/";
+    private static final String WRONG_PACKAGE_PATH = "com.bobocode.petros.wrong";
+    private static final String CORRECT_PACKAGE_PATH = "com.bobocode.petros.testpackage.scan.dependency";
 
     private AnnotationDependencyClassScanner sut;
 
     @BeforeEach
+    @SneakyThrows
     void setup() {
         sut = new AnnotationDependencyClassScanner();
+        var aClass = sut.getClass();
+        var isDebugMode = aClass.getDeclaredField("isDebugMode");
+        isDebugMode.setAccessible(true);
+        isDebugMode.set(sut, true);
+        var targetClassesPath = aClass.getDeclaredField("targetClassesPath");
+        targetClassesPath.setAccessible(true);
+        targetClassesPath.set(sut, TESTING_TEST_CLASSES_PATH);
+    }
+
+    @Test
+    void shouldThrowNoSuchPackageFoundExceptionWhenPathIsIncorrect() {
+        var expectedMessage = String.format("No package with name = %s found", WRONG_PACKAGE_PATH);
+        var noSuchPackageFoundException =
+                Assertions.assertThrows(NoSuchPackageFoundException.class, () -> sut.scan(WRONG_PACKAGE_PATH));
+        Assertions.assertEquals(expectedMessage, noSuchPackageFoundException.getMessage());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldThrowMultipleInjectConstructorsExceptionWhenClassHasTwoConstructorsMarkedInject() {
+        var strictClassPath = "com.bobocode.petros.testpackage.scan.dependency.TwoConstructorsMarkedInjectedCase";
+        setUpField(sut.getClass(), strictClassPath);
+        var expectedMessage = String.format("Class %s has more that one constructor with @Inject annotation", strictClassPath);
+        MultipleInjectConstructorsException multipleInjectConstructorsException =
```

Comment:
you also may use _var_ here, this will be cleaner

Reply:
Done!